app/components/admin/custom_fields/hierarchy/item_form_component.rb

```diff
@@ -29,14 +29,17 @@
 module Admin
   module CustomFields
     module Hierarchy
-      class NewItemFormComponent < ApplicationComponent
+      class ItemFormComponent < ApplicationComponent
         include OpTurbo::Streamable
 
-        def initialize(custom_field:, label: nil, short: nil)
+        def initialize(custom_field:, hierarchy_item:, url:, method:, label: nil, short: nil)
```

Comment:
Parameters look dirty, but at the same time seem necessary. I'm open to ideas on how to do this.